### PR TITLE
Reconcile configmap

### PR DESCRIPTION
**Pull request by ashleyschuett**
**What this PR does / why we need it**:
This PR is split into 3 parts by three commits. 

1. remove unused configmap reconcile code
2. reconcile on all labels/annotations that are required 
3. refactor code for readability 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Reconcile required labels and annotations on ConfigMap resources 
```

### CODE REVIEW

File: pkg/virt-operator/resource/apply/core.go [hunk 1]
```diff
@@ -463,145 +464,112 @@ func (r *Reconciler) createOrUpdateRbac() error {
 	return nil
 }
 
-func (r *Reconciler) createOrUpdateConfigMaps() error {
```
Comment (davidvossel) [inline pkg/virt-operator/resource/apply/core.go:466]: I understand this isn't used at the moment, but how do we explain that there's a list of ConfigMaps on the install strategy stucture, but we only reconcile one that's related to our CA. 

Also, what's the history of config map usage at install time. If we recently managed more configmaps, then we'd need to leave this function around for compatiblity when updating from old KubeVirt versions

Comment (ashleyschuett) [reply pkg/virt-operator/resource/apply/core.go:466]: As far as I can tell the configmap resource that is not a CA and the associated [sync code](https://github.com/kubevirt/kubevirt/pull/3133/files#diff-f5fa8e6ec51d9110d819dbb3eb3f3246cc221c958c0dfce503d269ac1516458dR661) was added in PR https://github.com/kubevirt/kubevirt/pull/3133 but that was then reverted in PR https://github.com/kubevirt/kubevirt/pull/3498. but the revert PR didn't include removing the sync loop.

Comment (davidvossel) [reply pkg/virt-operator/resource/apply/core.go:506]: great find. Yep that looks like it. And since that was never released, it's safe to remove the code.

**Review by davidvossel — APPROVED** [review]
/lgtm
/approve

**Pull request merged**